### ISSUE TITLE
Compiling fails with `fatal error: jni.h: No such file or directory`

### ISSUE THREAD

**Issue by WilliamIzzo83** [bazelbuild/rules_android_ndk#87]
I'm using Bazel 7.4.1 on Linux and I've followed the examples provided in Bazel documentation and `rules_android`/`rules_android_ndk` to do the setup. So I've installed the latest Android Studio (Ladybug | 2024.2.1 Patch 2) and I've downloaded the followings:

- SDK Platforms: Android 14.0 ("UpsideDownCake"), API Level 34
- SDK Tools: Android SDK Build-Tools 35.0.0
- SDK Tools: NDK (Side by side) 28.0.12674087
 
Both `$ANDROID_HOME` and `$ANDROID_NDK_HOME` are set accordingly as well as the required `.bazelrc` flags and `MODULE.bazel`. If I try to build an Android application without a `cc_library` dep, it works properly, but I can't seem to make Bazel compile an Android application with a `cc_library` dep.

Unfortunately I'm not able to make this work and I'm kinda lost here.
I'm attaching my workspace in the issue to troubleshoot the issue.

Thank you in advance.
[ndk-test.zip](https://github.com/user-attachments/files/17944615/ndk-test.zip)

**Comment by WilliamIzzo83** [comment]
After some tinkering I made it work. 

TL;DR 

Define the NDK platforms as shown in the [basic example](https://github.com/bazelbuild/rules_android_ndk/blob/main/examples/basic/BUILD.bazel) drop `--fat_apk_cpu` flag and build like so:

```
bazel build --android_platforms=:arm64-v8a,:x86 --andoird_crosstool_top=@androidndk//:toolchain
```
---
Long story version

The [example documentation](https://github.com/bazelbuild/rules_android_ndk/blob/main/examples/basic/README.md) says that to select the NDK toolchain you need to pass 

```
--fat_apk_cpu=arm64-v8a,x86 --android_crosstool_top=@androidndk//:toolchain
```

Doing so using `Bazel 7.4.1` with [rules_android](https://github.com/bazelbuild/rules_android/releases/tag/v0.5.1), [rules_android_ndk](https://github.com/bazelbuild/rules_android_ndk/releases/tag/v0.1.2) and `bzlmod` will fail with the error I've mentioned. So what I did is to take a look at how the CI actually [builds this repo](https://github.com/bazelbuild/rules_android_ndk/blob/main/.bazelci/presubmit.yml). Based on that what you need to do is to drop `--fat_apk_cpu` and use `--android_platforms` instead. 

To use the latter you need to define the platforms [somewhere](https://github.com/bazelbuild/rules_android_ndk/blob/main/examples/basic/BUILD.bazel) and pass them to the flag. Therefore suppose you've defined them in the root of your workspace as shown above the correct invocation of Bazel build would be

```
bazel build --android_platforms=:arm64-v8a,:x86 --andoird_crosstool_top=@androidndk//:toolchain
```

At this point the apk is built.

### Final note
I don't know if all of this is already common knowledge, but at least to me starting anew the documentations, both here and Bazel's, really dropped me off. Would maintainers here accept a contribution to fix the situation at least here?

**Comment by ted-xie** [comment]
This should be the minimal set of flags to build the example app (i.e. this is what CI does): https://github.com/bazelbuild/rules_android_ndk/blob/main/.bazelci/presubmit.yml#L8-L12

I think even some of those aren't strictly necessary anymore; it should be sufficient to pass `--android_platforms=//:arm64-v8a,//:x86`. I don't think --android_crosstool_top should be necessary.

Yeah the documentation is pretty out of date :(. We do plan on fixing and improving documentation for rules_android and rules_android_ndk in the near future. In the mean time, if you want to update the documentation to what you think should be more accurate, then we'd be open to reviewing contributions.
